src/libs/decoders/mp3.cpp

```diff
@@ -159,9 +160,9 @@ static Sint32 MP3_seek(Sound_Sample* const sample, const Uint32 ms)
 {
     Sound_SampleInternal* const internal = static_cast<Sound_SampleInternal*>(sample->opaque);
     mp3_t* p_mp3 = static_cast<mp3_t*>(internal->decoder_private);
-    const float frames_per_ms = sample->actual.rate / 1000.0f;
-    const drmp3_uint64 frame_offset = static_cast<drmp3_uint64>(frames_per_ms) * ms;
-    const Sint32 result = drmp3_seek_to_pcm_frame(p_mp3->p_dr, frame_offset);
+    const uint64_t sample_rate = sample->actual.rate;
+    const drmp3_uint64 pcm_frame = ceil_divide(sample_rate * ms, 1000u);
```

Comment:
There's a bug in the original `sample->actual.rate / 1000.0f` when dividing the `int / float`: `44100 / 1000.0f` becomes `44.0`; so seeks were slightly off when using 44.1 KHz MP3s; usually only noticeable in games that used a long single track and seeked to different segments within, such as Stellar 7. (When encoding with `lame -V7`, the sampling rate becomes 32,000 Hz and above `-V7` lame uses 24,000; so both of these were unaffected).

The integer division really tidies up the code, and lands the exact PCM frame without any decimal point or casting nonsense. (Longer term, all the audio code should use this call; but that's for a future PR).